genie_registry/clinical.py

```diff
@@ -77,6 +77,92 @@ def _check_year(clinicaldf: pd.DataFrame, year_col: int, filename: str,
+        Error message if values and inconsistent or blank string
+    """
+    cols = ["INT_DOD", "DEAD"]
+    for col in cols:
+        # Return empty string is columns don't exist because this error
+        # is already handled.
+        if not process_functions.checkColExist(clinicaldf, col):
+            return ''
+    is_dead = clinicaldf['DEAD'].astype(str) == "True"
+    is_alive = clinicaldf['DEAD'].astype(str) == "False"
+    allowed_str = ['Unknown', 'Not Collected', 'Not Applicable',
+                   "Not Released"]
+    is_str = clinicaldf['DEAD'].isin(allowed_str)
+    # Check that all string values are equal each other
+    is_equal = all(
+        clinicaldf.loc[is_str, "DEAD"] == clinicaldf.loc[is_str, "INT_DOD"]
+    )
+    # If dead, int column can't be Not Applicable
+    # If alive, int column can't have values
+    if (any(clinicaldf.loc[is_dead, 'INT_DOD'] == "Not Applicable") or
+            not all(clinicaldf.loc[is_alive, 'INT_DOD'].isin(allowed_str)) or
+            not is_equal):
+        return (
+            "Patient: you have inconsistent values in "
+            "INT_DOD, YEAR_DEATH, DEAD\n"
+        )
+    return ''
+
+
+def _check_int_year_consistency(
+        clinicaldf: pd.DataFrame,
+        cols: list, string_vals: list
+    ) ->  str:
+    """
+    Check if vital status interval and year columns are consistent in
+    their values
+
+    Args:
+        clinicaldf: Clinical Data Frame
+        cols: Columns in the clinical data frame
+        string_vals: String values that aren't integers
+
+    Returns:
+        Error message if values and inconsistent or blank string
+    """
+    interval_col = ''
+    year_col = ''
+    for col in cols:
+        # This is assuming that interval and year columns start with
+        # INT/YEAR
+        interval_col = col if col.startswith("INT") else interval_col
+        year_col = col if col.startswith("YEAR") else year_col
+        # Return empty string is columns don't exist because this error
+        # is already handled.
+        if not process_functions.checkColExist(clinicaldf, col):
+            return ""
+
+    is_inconsistent = False
+    # Get index of all rows that have 'missing' values
+    for str_val in string_vals:
+        # n string values per row
+        n_str = (clinicaldf[cols] == str_val).sum(axis=1)
+        if n_str.between(0, len(cols), inclusive="neither").any():
+            is_inconsistent = True
+
+    # Check that the redacted values are consistent
+    is_redacted_int_89 = clinicaldf[interval_col] == ">32485"
+    is_redacted_year_89 = clinicaldf[year_col] == ">89"
+    is_redacted_int = clinicaldf[interval_col] == "<6570"
+    is_redacted_year = clinicaldf[year_col] == "<18"
+    if (any(is_redacted_int != is_redacted_year) or
+            any(is_redacted_int_89 != is_redacted_year_89)):
+        is_inconsistent = True
+
+    if is_inconsistent:
+        return f"Patient: you have inconsistent values in {', '.join(cols)}\n"
```

Comment:
Similar to identifying inconsistencies with the DEAD variable, it would be great if the error message could include the patient IDs for rows with inconsistent values to facilitate identification of these rows.

f"Patient Clinical File: you have inconsistent values in {', '.join(cols)} for the following patients: GENIE-xxx-xxxx, GENIE-xxx-xxxx\n"